modules/nf-core/hifiasm/main.nf

```diff
@@ -32,6 +35,8 @@ process HIFIASM {
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def ploidy_value = ploidy ? "--n-hap $ploidy" : ""
+    def gen_size_kb_value = gen_size_kb ? "--hg-size \${gen_size_kb}k" : ""
```

Comment:
You only need to escape `$` when you want to reference a variable in bash context.